src/dune_pkg/opamUrl0.mli

```diff
@@ -15,10 +15,11 @@ val is_version_control : t -> bool
 (** [is_file t] is true iff [t] is a url beginning with "file://" *)
 val is_local : t -> bool
 
-(* [local_or_git_only t loc] returns [`Path p] for a URL pointing to a local
-   file system or [`Git] if it's a git repository (remote or otherwise). If
-   it's neither of those cases, it will error out. *)
-val local_or_git_only : t -> Loc.t -> [ `Path of Path.t | `Git ]
+(* [local_or_git_or_tar_only t loc] returns [`Path p] for a URL pointing to a
+   local file system or [`Git] if it's a git repository (remote or otherwise)
+   or [`Tar] if it's an http server with a tar resource. If it's neither of
+   those cases, it will error out. *)
+val local_or_git_or_tar_only : t -> Loc.t -> [ `Path of Path.t | `Git | `Tar ]
```

Comment:
I think I remembered it but can you explain again why `Tar` is not `Tar of Path.t` where the path would be the path to the extracted source?

Reply:
Because Tar is transformed to a `Path` where I need to use it, and we compute the path afterward, not in this function as we don't know where the tarball will be extracted yet.